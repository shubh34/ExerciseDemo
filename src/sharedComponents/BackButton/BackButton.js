/* eslint-disable max-len */
import './BackButton.scss';
import React from 'react';
import PropTypes from 'prop-types';

const BackButton = ({ onClick, title = 'Back' }) => (
	<div className="back-button">
		<button type="button" name="go back" onClick={onClick}>{title}</button>
	</div>

);
BackButton.propTypes = {
	onClick: PropTypes.string.isRequired,
	title: PropTypes.string,
};

export default BackButton;
