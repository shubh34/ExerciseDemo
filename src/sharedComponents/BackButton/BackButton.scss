.back-button {
    z-index: 110;
    position: absolute; 
   & :before {
       content: '\e628';
       font-family: Glyphs;
       font-size: 32px;
   };
   button {
    font-size: 24px;
    padding: 16px;
    background: transparent;
    border: none;
    color:white;
    cursor: pointer;
   }
}