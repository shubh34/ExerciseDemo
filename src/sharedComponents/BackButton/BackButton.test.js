import React from 'react';
import { shallow } from 'enzyme';
import BackButton from './BackButton';

describe('BackButton', () => {
	it('renders without crashing', () => {
		const wrapper = shallow(<BackButton title='xyz' onClick={jest.fn()} />);
		expect(wrapper).toMatchSnapshot();
	});
	it('should call onClick on button click', () => {
		const onClick = jest.fn();
		const wrapper = shallow(<BackButton title='xyz' onClick={onClick} />);
		wrapper.find('button').simulate('click');
		expect(onClick).toBeCalled();
	});
});
