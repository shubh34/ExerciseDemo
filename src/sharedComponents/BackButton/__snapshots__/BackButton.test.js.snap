// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton renders without crashing 1`] = `
<div
  className="back-button"
>
  <button
    name="go back"
    onClick={[MockFunction]}
    type="button"
  >
    xyz
  </button>
</div>
`;
